src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1625,97 +1649,235 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
+                // Flow the changes made to the authentication ticket.
+                ticket = endpointContext.AuthenticationTicket;
 
-            // Add the claims extracted from the access token.
-            foreach (var claim in ticket.Identity.Claims) {
-                notification.Claims.Add(claim);
-            }
+                if (endpointContext.HandledResponse)
+                {
+                    return;
+                }
 
-            await Options.Provider.ValidationEndpoint(notification);
+                if (ticket.Properties.ExpiresUtc.HasValue && ticket.Properties.ExpiresUtc < Options.SystemClock.UtcNow)
+                {
+                    Options.Logger.WriteVerbose("expired token");
+                    payload.Add("active", false);
+                }
 
-            // Flow the changes made to the authentication ticket.
-            ticket = notification.AuthenticationTicket;
+                //TODO: do more testing for token validity.
 
-            if (notification.HandledResponse) {
-                return;
-            }
+                else
+                {
+                    payload.Add(OpenIdConnectConstants.Claims.Active, true);
 
-            var payload = new JObject();
+                    var scopes = ticket.GetScopes();
+                    if (scopes.Any())
+                    {
+                        payload.Add(OpenIdConnectConstants.Claims.Scope, string.Join(" ", scopes));
+                    }
+                    
+                    payload.Add(OpenIdConnectConstants.Claims.ClientId, request.ClientId);
+
+                    // TODO: Re-examine whether ticket.Identity will always be the resource owner who authorized the token
+                    // and whether Username should always be returned
+                    payload.Add(OpenIdConnectConstants.Claims.Username, ticket.Identity.Name);
+
+                    // TODO: Determine the token type for the given token when not an access token
+                    // See https://tools.ietf.org/html/rfc7662#section-2.2 and https://tools.ietf.org/html/rfc6749#section-5.1
+                    if (hint.Equals(OpenIdConnectConstants.TokenHintTypes.AccessToken))
+                    {
+                        payload.Add(OpenIdConnectConstants.Claims.TokenType, "Bearer");
+                    }
 
-            payload.Add("audiences", JArray.FromObject(ticket.Properties.GetAudiences()));
-            payload.Add("expires_in", ticket.Properties.ExpiresUtc.Value);
-            
-            payload.Add("claims", JArray.FromObject(
-                from claim in notification.Claims
-                select new { type = claim.Type, value = claim.Value }
-            ));
+                    if (ticket.Properties.ExpiresUtc.HasValue)
+                    {
+                        payload.Add(JwtRegisteredClaimNames.Exp, ticket.Properties.ExpiresUtc.Value.Ticks);
```

Comment:
Why not simply using `IssuedUtc/ExpiresUtc.Value.UtcDateTime` as we already do when creating the `iat` claim?

https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/vNext/src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Tokens.cs#L325-L326